pkg/policies/branch/branch.go

```diff
@@ -267,6 +288,42 @@ func check(ctx context.Context, rep repositories, c *github.Client, owner,
 				d.BlockForce = false
 			}
 		}
+		rsc := p.GetRequiredStatusChecks()
+		if rsc != nil {
+			d.RequireUpToDateBranch = rsc.Strict
+			if mc.RequireUpToDateBranch && !rsc.Strict {
+				text = text +
+					fmt.Sprintf("Require up to date branch not configured for branch %v\n",
+						b)
+				pass = false
+			}
+			d.RequireStatusChecks = rsc.Contexts
+			c := make(map[string]struct{}, len(rsc.Contexts))
+			for _, ctx := range rsc.Contexts {
+				c[ctx] = struct{}{}
+			}
+			for _, check := range mc.RequireStatusChecks {
+				if _, ok := c[check]; !ok {
+					text = text +
+						fmt.Sprintf("Status check %v not required for branch %v\n",
+							check, b)
+					pass = false
+				}
+			}
+		} else {
+			d.RequireUpToDateBranch = false
+			if mc.RequireUpToDateBranch {
```

Comment:
So, I'd think we remove this case, if we make the above change that this is only applicable if `len(mc.RequireStatusChecks) > 0`